bindings/python/Cargo.toml

```diff
@@ -0,0 +1,12 @@
+[package]
+name = "opendal-python"
+version = "0.0.1"
+edition = "2021"
+
+[lib]
+name = "opendal"
```

Comment:
Now we're building C bindings, but the name `-lopendal` is occupied by python bindings.

@Xuanwo thoughts? Shall we choose one bindings to use `opendal.so` or leave every binding as `opendal_xxx.so`?

Reply:
pyo3 requires the lib name to be opendal if we want to use `import opendal` in python directly. I will visit @messense for some advice.

Reply:
For pure Rust project layout, the `.so` file name has to be `opendal` because that's how Python `import` works. We can switch to mixed Python/Rust layout to change it to whatever we want.